src/Validator.php

```diff
@@ -0,0 +1,63 @@
+<?php
+/**
+ * This file is part of Lcobucci\JWT, a simple library to handle JWT and JWS
+ *
+ * @license http://opensource.org/licenses/BSD-3-Clause BSD-3-Clause
+ */
+
+namespace Lcobucci\JWT;
+
+use Generator;
+use Lcobucci\JWT\Claim\Validatable;
+
+/**
+ * Class that validates token with defined validationData
+ *
+ * @author Danny Dörfel <danny.dorfel@gmail.com>
+ */
+class Validator
+{
+    protected $errors = [];
+
+    /**
+     * @param Token $token
+     * @param ValidationData $data
+     * @return bool
+     */
+    public function validate(Token $token, ValidationData $data): bool
+    {
+        $this->errors = [];
```

Comment:
I thought about creating a JWT rule on Respect\Validation sometimes. Of course we can do that but we need to write too much code from this library on our library then the chances of breaking the API is higher than using a single point of failure, that's why this feature is so useful. Also I don't thing it's fair to force someone to install one more library to have a fature that's simple for to have here.

About this implementation, IMHO, I would say that it can be better and the problem is on this line. If is there something I would get away from is to change the state of the object on each validation, and that what is happening here.

First of all, this method breaks SRP and secondly there is no need for such thing, not here. This method should only return `true` or `false` and that's all.

So, how can we get the errors and still not change the state of the object?

You can do what we do on Validation: Have a method to `assert()` the claims, throwing an exception when one claim or more are invalid then reuse the same method on `validate()` with a try/catch statement and return `true` if not exception is thrown or `false` if an exception is thrown, so you don't break DRY.